pkg/api/annotations_test.go

```diff
@@ -763,6 +788,144 @@ func TestService_AnnotationTypeScopeResolver(t *testing.T) {
 	}
 }
 
+func TestAPI_MassDeleteAnnotations_AccessControl(t *testing.T) {
+	sc := setupHTTPServer(t, true, true)
+	setInitCtxSignedInEditor(sc.initCtx)
+	_, err := sc.db.CreateOrgWithMember("TestOrg", testUserID)
+	require.NoError(t, err)
+
+	type args struct {
+		permissions []*accesscontrol.Permission
+		url         string
+		body        io.Reader
+		method      string
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want int
+	}{
+		{
+			name: "Mass delete dashboard annotations without dashboardId is not allowed",
+			args: args{
+				permissions: []*accesscontrol.Permission{{Action: accesscontrol.ActionAnnotationsDelete, Scope: accesscontrol.ScopeAnnotationsTypeOrganization}},
+				url:         "/api/annotations/mass-delete",
+				method:      http.MethodPost,
+				body: mockRequestBody(dtos.MassDeleteAnnotationsCmd{
+					DashboardId: 0,
+					PanelId:     1,
+				}),
+			},
+			want: http.StatusBadRequest,
+		},
+		{
+			name: "Mass delete dashboard annotations without panelId is not allowed",
+			args: args{
+				permissions: []*accesscontrol.Permission{{Action: accesscontrol.ActionAnnotationsDelete, Scope: accesscontrol.ScopeAnnotationsTypeOrganization}},
+				url:         "/api/annotations/mass-delete",
+				method:      http.MethodPost,
+				body: mockRequestBody(dtos.MassDeleteAnnotationsCmd{
+					DashboardId: 10,
+					PanelId:     0,
+				}),
+			},
+			want: http.StatusBadRequest,
+		},
+		{
+			name: "AccessControl mass delete dashboard annotations with correct dashboardId and panelId as input is allowed",
+			args: args{
+				permissions: []*accesscontrol.Permission{{Action: accesscontrol.ActionAnnotationsDelete, Scope: accesscontrol.ScopeAnnotationsTypeDashboard}},
+				url:         "/api/annotations/mass-delete",
+				method:      http.MethodPost,
+				body: mockRequestBody(dtos.MassDeleteAnnotationsCmd{
+					DashboardId: 1,
+					PanelId:     1,
+				}),
+			},
+			want: http.StatusOK,
+		},
+		{
+			name: "Mass delete organization annotations without input to delete all organization annotations is allowed",
```

Comment:
It could be used in weird ways. Maybe we should define, code and document this endpoint but at this point this is how it works. I wanted to focus on FGAC changes and not change the behavior but we definitely need to address this endpoint in the future.